DataFormats/Detectors/Common/include/DetectorsCommonDataFormats/EncodedBlocks.h

```diff
@@ -874,22 +891,20 @@ void EncodedBlocks<H, N, W>::encode(const input_IT srcBegin,      // iterator be
                              static_cast<int32_t>(frequencyTable.size()),
                              dataSize,
                              static_cast<int32_t>(nLiteralSymbols)};
-
   } else { // store original data w/o EEncoding
-    const size_t messageSizeBytes = messageLength * sizeof(input_t);
-    const int dataSize = messageSizeBytes / sizeof(storageBuffer_t) + (sizeof(input_t) < sizeof(storageBuffer_t));
-    // no dictionary needed
-    expandStorage(dataSize);
-    const size_t nPaddingElements = messageSizeBytes % sizeof(storageBuffer_t);
     //FIXME(milettri): we should be able to do without an intermediate vector;
     // provided iterator is not necessarily pointer, need to use intermediate vector!!!
     std::vector<input_t> vtmp(srcBegin, srcEnd);
+    // introduce padding in case literals don't align;
+    const size_t nPaddingElements = calculatePadding<input_t, storageBuffer_t>(messageLength);
     vtmp.insert(vtmp.end(), nPaddingElements, {});
```

Comment:
you still have 1 guaranteed re-allocation of vtmp here, why not booking it from very beginning with the right size?

Reply:
see comment below.

Reply:
which comment? This `//FIXME(milettri): we should be able to do without an intermediate vector`? 
Are you going to change the storeData interface?